src/DDTrace/Integrations/CodeIgniter/V2/CodeIgniterSandboxedIntegration.php

```diff
@@ -0,0 +1,212 @@
+<?php
+
+namespace DDTrace\Integrations\CodeIgniter\V2_2;
+
+use DDTrace\Configuration;
+use DDTrace\Contracts\Span;
+use DDTrace\GlobalTracer;
+use DDTrace\Integrations\SandboxedIntegration;
+use DDTrace\SpanData;
+use DDTrace\Tag;
+use DDTrace\Type;
+
+class CodeIgniterSandboxedIntegration extends SandboxedIntegration
+{
+    const NAME = 'codeigniter';
+
+    /**
+     * @return string The integration name.
+     */
+    public function getName()
+    {
+        return self::NAME;
+    }
+
+    /**
+     * Add instrumentation to CodeIgniter requests
+     */
+    public function init()
+    {
+
+        $tracer = GlobalTracer::get();
+        if (!$tracer) {
```

Comment:
When will there not be a root scope, btw?

Reply:
This can occur when the tracer does not get bootstrapped and does not create a root span (and thus no root scope). Also in this case the `NoopTracer` get's used. :)